[#5376] Improve conversion functions, auto-convert weights

arbron opened the pull request:
Extends `convertLength` and `convertWeight` to support automatic target unit selection similar to `convertTime`, and adds the ability for all three of those methods to automatically select the equivalent unit in another measurement system (e.g. mi => km).

This change is then used during item preparation to automatically convert item weights into the proper measurement system depending on the "Use Metric Weight Units" option. Since this is done at data preparation rather than sheet rendering, this setting now requires a restart.

 ### Todo
- [ ] Automatically convert length units

arbron committed:
[#5376] Improve conversion functions, auto-convert weights

Extends `convertLength` and `convertWeight` to support automatic
target unit selection similar to `convertTime`, and adds the
ability for all three of those methods to automatically select the
equivalent unit in another measurement system (e.g. mi => km).

This change is then used during item preparation to automatically
convert item weights into the proper measurement system depending
on the "Use Metric Weight Units" option. Since this is done at
data preparation rather than sheet rendering, this setting now
requires a restart.

 ### Todo
- [ ] Automatically convert length units

pgervais70 commented:
Could it be possible also to expose the convertLength and convertWeight functions to be used by macros or modules?

JPMeehan commented:
> Could it be possible also to expose the convertLength and convertWeight functions to be used by macros or modules?

pretty sure they would be available under dnd5e.utils

arbron committed:
[#5376] Fix issue with unit selection, move to separate method

Switches to using the conversion multiplier magnitude as a more
reliable method for selecting the closes unit in the preferred
system. Explit the method that selects that unit into its own
method so it can be used elsewhere.
